### PR TITLE
[BACKLOG-6092] Fixed api.operation.state for autoSubmit flag and fixe…

### PR DESCRIPTION
**Pull request by annapopova1**
…d some js doc

@diogofscmariano @krivera-pentaho @pamval @plagoa @jvelasques @afrjorge please review

### CODE REVIEW

File: package-res/resources/web/prompting/PromptPanel.js [hunk 1]
```diff
@@ -43,6 +43,7 @@ define(['cdf/lib/Base', 'cdf/Logger', 'dojo/number', 'dojo/i18n', 'common-ui/uti
           incorrectBooleanType: function(name, value) {
             return "Unexpected value '" + value + "' for '" + name + "'. Must be boolean type.";
           },
+          notAllowedAutoSubmit: "Not possible to set 'autoSubmit'. It's limeted by the 'allowAutoSubmit' flag",
```
Comment (afrjorge) [inline package-res/resources/web/prompting/PromptPanel.js]: typo "limited"

File: package-res/resources/web/prompting/PromptPanel.js [hunk 2]
```diff
@@ -354,13 +355,13 @@ define(['cdf/lib/Base', 'cdf/Logger', 'dojo/number', 'dojo/i18n', 'common-ui/uti
       };
 
       /**
-       * Checks input state parameter to contain read only properties. If conatins, it throws an exception
+       * Checks input state parameter to contain read only properties. If conatins, it throws an exception.
```
Comment (afrjorge) [inline package-res/resources/web/prompting/PromptPanel.js]: typo "contains"

**Comment by annapopova1** [comment]
@afrjorge thanks (fixed).

File: package-res/resources/web/prompting/PromptPanel.js [hunk 3]
```diff
@@ -395,8 +411,10 @@ define(['cdf/lib/Base', 'cdf/Logger', 'dojo/number', 'dojo/i18n', 'common-ui/uti
        * @name PromptPanel#_validateStatePage
        * @method
        * @private
-       * @param  {Object} value The value of page
-       * @throws {String}       Exception if 'page' is incorrect
+       * @param {Number} page       The value of page
+       * @param {Boolean} paginate  The whether pagination is active
```
Comment (afrjorge) [inline package-res/resources/web/prompting/PromptPanel.js]: How about something like "A flag indicating if pagination is active"?

File: package-res/resources/web/prompting/PromptPanel.js [hunk 4]
```diff
@@ -1542,12 +1561,12 @@ define(['cdf/lib/Base', 'cdf/Logger', 'dojo/number', 'dojo/i18n', 'common-ui/uti
         /**
          * Modifys a state of the prompting system.
          *
-         * @name PromptPanel@setState
+         * @name PromptPanel#setState
          * @method
-         * @param {Object} [state]                    The set of flags which will be applyed to current state. It's optional parameter.
+         * @param {Object} state                      The set of flags which will be applyed to current state.
```
Comment (afrjorge) [inline package-res/resources/web/prompting/PromptPanel.js]: typo "applied"

File: package-res/resources/web/test/prompting/PromptPanelSpec.js [hunk 5]
```diff
@@ -1326,6 +1326,17 @@ define([ 'dojo/number', 'dojo/i18n', 'common-ui/prompting/PromptPanel',
...
 
+        it("should throw exception if try set not allowed autoSubmit flag", function() {
+          var paramDefnSpy = jasmine.createSpyObj("paramDefnSpy", [ "allowAutoSubmit" ]);
+          spyOn(panel, "getParamDefn").and.returnValue(paramDefnSpy);
+          expect(function() {
+            var state = {
+              autoSubmit: true
+            };
+            panel.setState(state);
+          }).toThrow("Not possible to set 'autoSubmit'. It's limeted by the 'allowAutoSubmit' flag");
```
Comment (afrjorge) [inline package-res/resources/web/test/prompting/PromptPanelSpec.js]: same typo "limited"

**Comment by krivera-pentaho** [comment]
Looks good to me

**Pull request merged**